Deaths doors Tweakings and misc stuff.

Aurorablade opened the pull request:
## About The Pull Request
HI! this touches some issues with deaths doors, anything more complex may need to pass onto Reen when they are not eeeppyy

Fix break down 1: The initial reopen proc to reopen the portal..just made a new portal and caused shinnigans...IT NEEDED A RETURN ARRHHHH
Issue: its still not moving the previous portal out of the casters 'contents' but it IS spiting out the mob.

Fix breakdown 2: IT SHOULD spit out the mob properly if you do not reopen a portal after closing it and the mob will not be stuck in the caster.. SHOULD... **SHOULD**.

Fix breakdown 3: Just changed some namings of a var and a order cause it made sense to me, also TIMER_STOPPABLE cause it was throwing a big ol fit.

Addition: You can now RIGHT CLICK the portal to eject the mob early.

This hopefully will be able to give reen a good start when he not eepy (Trapped might could be a list imo, just so we don't run into trouble storeing more then one thing.

Additions to make me happy:
DO YOU LIKE CODER SPRITES?! I DO
Necras sight now has a poorly resprited spooky eyeball as a spell icon. instead of a psycross. ![](https://puu.sh/Kw6Z4/54d670c062.png)

and now borrowed time has a icon cause i am tired of pretending its okay it didn't have one.
![](https://puu.sh/Kw6Z7/07c10fd384.png)
<!-- Describe your pull request. Avoid text walls, use concise bullet points for easier readability. Document every change, or this can delay review and even discourage maintainers from merging your PR. -->

## Testing Evidence
i forgot to take a gif recording did test it...
<!-- It's mandatory to test your PR. Provide images, clips or description of how you tested your changes where possible. -->

## Why It's Good For The Game
PRAISE THE UNDERMAIDEN
<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial. If you can't, then it probably isn't good for the game in the first place. -->

Aurorablade commented:
Me and reen were talking about it in the discord bug channel as well

Aurorablade commented:
Fethas note to self..try a check and logic switch on the portal reopening logic. I need to see what the code thinks its doing vs what i think its doing.

reen if you read this freel free to shove edits my way.

chemistrymain3 commented:
works

@Aurorablade remember to also add a spitout function to destroy() as well i was going to do that but i'm getting wrapped up in trying to figure out how to do something that's not content.add for this instead

otherwise whoever is in the portal gets qdel'd on destroy

honestly as simple as like

/obj/structure/underworld_portal/Destroy()
	visible_message(span_danger("The portal collapses with an angry hiss."))
	spitout_mob(caster, loc)
	..()

Aurorablade commented:
am working on this just might not do a commit tonight.

Aurorablade commented:
Changed the sound to the deadbell then a loud ass churn felt more thematic..I could have also found some creepy whispers.

IcOn MeRge ConFlicts

added a furhter check for the dispelled logic.

Dystroyed check for if anything BESIDES THE CASTER removes the portal from existance.

The portal now respawns correctly it needed to be if(src in contents) not if(loc in contents) I had initally thought that as loc is checking for the existance of a location not an object in a thing. This is the one cause the implied src needs to be FURTHER implied.
